presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -158,13 +158,13 @@ public static HiveSplitSource allAtOnce(
                     }
 
                     @Override
-                    public void finish()
+                    public void finishSplitLoading()
                     {
                         queue.finish();
                     }
 
                     @Override
-                    public boolean isFinished(OptionalInt bucketNumber)
+                    public boolean isSplitFetchingFinished(OptionalInt bucketNumber)
```

Comment:
Personally i would keep the name `isFinished(bucketNumber)`. It seems like it clearly indicates that there will be no more splits for this bucket (unless rewound). 

It feels like the answer to the `isSplitFetchingFinished` is engine specific. Like `if the engine  has finished fetching splits`. What this method is intended to give the answer for is rather `are there any more splits to be fetch`. However i don't know how to express latest in concise manner. Thus I'm advocating for `isFinished`. Thoughts?

Reply:
Good point. I think if we rename `finish` to `noMoreSplits`, then `isFinished` by itself is not that confusing, so I'll leave that unchanged.